File: frontend/src/types/exam.ts
export type ExamStatus = "Scheduled" | "Arrived" | "Canceled" | "Completed";

export interface ExamFormValues {
  patientId: string;
  patientName: string;
  gender: "Male" | "Female";
  birthdate: string;
  email: string;
  examType: string;
  dateTime: string;
  comments?: string;
  status: ExamStatus;
}

export interface ExamFilter {
  patientId?: string;
  patientName?: string;
  gender?: string;
  birthdate?: string; // ISO format (YYYY-MM-DD)
  email?: string;
  examType?: string;
  examDateFrom?: string; // ISO date
  examDateTo?: string; // ISO date
  status?: string;
}

export interface Exam extends ExamFormValues {
  id: number;
}

File: backend/Models/Exam.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace backend.Models
{
    public class Exam
    {
        public int Id { get; set; }

        // Patient Details
        [Required]
        public string PatientId { get; set; } = null!;

        [Required]
        public string PatientName { get; set; } = null!;

        [Required]
        [RegularExpression("Male|Female|Other", ErrorMessage = "Gender must be Male, Female, or Other")]
        public string Gender { get; set; } = null!;

        [Required]
        public DateTime Birthdate { get; set; }

        [Required]
        [EmailAddress]
        public string Email { get; set; } = null!;

        // Exam Details
        [Required]
        public string ExamType { get; set; } = null!;  // e.g., "CT Brain", "Chest X-Ray"

        [Required]
        public DateTime ExamDate { get; set; }         // Includes both date & time

        public string? Comments { get; set; }

        // Status
        [Required]
        [RegularExpression("Scheduled|Arrived|Canceled|Completed", ErrorMessage = "Invalid status")]
        public string Status { get; set; } = null!;
    }
}
